rest/upgradetest/upgrade_registry_test.go

```diff
@@ -160,6 +162,171 @@ func TestUpgradeDatabasePreHelium(t *testing.T) {
+	require.NoError(t, err)
+
+	dbConfig, ok := dbMap["db"]
+	require.True(t, ok)
+
+	// Need to sanitize the db config, but can't use sanitizeDbConfigs because it assumes non-empty server address
+	dbConfig.Username = ""
+	dbConfig.Password = ""
+	dbConfigBytes, err := base.JSONMarshal(dbConfig)
+
+	require.NoError(t, err)
+	resp = persistentRT.SendAdminRequest("PUT", "/db/", string(dbConfigBytes))
+	assert.Equal(t, http.StatusCreated, resp.Code)
+
+	// check if database is online
+	dbRoot := persistentRT.GetDatabaseRoot("db")
+	require.Equal(t, db.RunStateString[db.DBOnline], dbRoot.State)
+}
+
+// TestMetadataIDRenameDatabase verifies that resync is not required when deleting and recreating a database (with a
+// different name) targeting only the default collection.
+func TestMetadataIDRenameDatabase(t *testing.T) {
+
+	if base.TestsUseNamedCollections() {
+		t.Skip("This test covers legacy interaction with the default collection")
+	}
+
+	// Create a persistent rest tester.
+	rt := rest.NewRestTester(t, &rest.RestTesterConfig{
+		PersistentConfig: true,
+	})
+	defer rt.Close()
+
+	dbConfig := rt.NewDbConfig()
+	dbName := "db"
+	rt.CreateDatabase(dbName, dbConfig)
+
+	// Write a document to ensure _sync:seq is initialized
+	resp := rt.SendAdminRequest("PUT", "/"+dbName+"/testRenameDatabase", `{"test":"test"}`)
+	require.Equal(t, http.StatusCreated, resp.Code)
+
+	// Delete database
+	resp = rt.SendAdminRequest("DELETE", "/"+dbName+"/", "")
+	require.Equal(t, http.StatusOK, resp.Code)
+
+	newDbName := "newdb"
+	dbConfig.Name = newDbName
+
+	// Create a new db targeting the same bucket
+	rt.CreateDatabase(newDbName, dbConfig)
+
+	// check if database is online
+	dbRoot := rt.GetDatabaseRoot(newDbName)
+	require.Equal(t, db.RunStateString[db.DBOnline], dbRoot.State)
+}
+
+// Verifies that matching metadataIDs are computed if two config groups for the same database are upgraded
+func TestMetadataIDWithConfigGroups(t *testing.T) {
+
+	if base.TestsUseNamedCollections() {
+		t.Skip("This test covers legacy interaction with the default collection")
+	}
+
+	testCtx := base.TestCtx(t)
+
+	tb1 := base.GetPersistentTestBucket(t)
+	// Create a non-persistent rest tester.  Standard RestTester
+	// creates a database 'db' targeting the default collection (when !TestUseNamedCollections)
+	legacyRT := rest.NewRestTester(t, &rest.RestTesterConfig{
+		CustomTestBucket: tb1.NoCloseClone(),
+		PersistentConfig: false,
+	})
+
+	// Create a document in the collection to trigger creation of _sync:seq
+	resp := legacyRT.SendAdminRequest("PUT", "/db/testLegacyMetadataID", `{"test":"test"}`)
+	assert.Equal(t, http.StatusCreated, resp.Code)
+
+	// Get the legacy config for upgrade test below
+	resp = legacyRT.SendAdminRequest("GET", "/_config?include_runtime=true", "")
+	legacyConfigBytes := resp.BodyBytes()
+
+	var legacyConfig rest.LegacyServerConfig
+	err := base.JSONUnmarshal(legacyConfigBytes, &legacyConfig)
+	assert.NoError(t, err)
+	legacyRT.Close()
```

Comment:
move this up to defer at the top of test to protect other tests?

Reply:
This needs to be closed before opening the non-legacy RTs, so leaving as-is.